packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java

```diff
@@ -196,13 +214,22 @@ public ResponseEntity<String> serveEndpoint(
                             EndpointAccessChecker.ACCESS_DENIED_MSG));
         }
 
+        // Put a VaadinRequest in the instances object so as the request is
+        // available in the endpoint method
+        VaadinRequest vaadinRequest = createVaadinRequest(request);
+        VaadinService service = vaadinRequest.getService();
+        VaadinResponse vaadinResponse = (httpServletResponse != null)
+                ? new VaadinServletResponse(httpServletResponse,
+                        (VaadinServletService) service)
+                : null;
+
         try {
-            // Put a VaadinRequest in the instances object so as the request is
-            // available in the endpoint method
-            VaadinServletService service = (VaadinServletService) VaadinService
-                    .getCurrent();
-            CurrentInstance.set(VaadinRequest.class,
-                    new VaadinServletRequest(request, service));
+            if (service != null) {
+                DAUUtils.TrackableOperation.INSTANCE.execute(() -> {
+                    service.requestStart(vaadinRequest, vaadinResponse);
+                    DAUUtils.applyEnforcement(vaadinRequest, unused -> true);
```

Comment:
Once the Flow PR will be merged, this should be changed int this way
```java
    if (DAUUtils.isDauEnabled(service)) {
        DAUUtils.applyEnforcement(vaadinRequest, unused -> true)
    }
```

Reply:
Done.